### PR TITLE
Interrupt test if network changed

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Fixes https://github.com/ooni/probe/issues/1296

## Proposed Changes

  - Interrupt test when connectivity changes

### CODE REVIEW

File: ooniprobe/AppDelegate.m [hunk 1]
```diff
@@ -28,6 +29,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     [[UIBarButtonItem appearance] setTitleTextAttributes:[NSDictionary dictionaryWithObjectsAndKeys: [UIFont fontWithName:@"FiraSans-Regular" size:16],NSFontAttributeName, nil] forState:UIControlStateNormal];
     [NavigationBarUtility setDefaults];
     [CountlyUtility initCountly];
+    [ReachabilityManager sharedManager];
```
Comment (bassosimone) [inline ooniprobe/AppDelegate.m:32]: Is the purpose of this statement to create a singleton instance? If so, could you please document it?

File: ooniprobe/View/RunTest/TestRunningViewController.m [hunk 2]
```diff
@@ -169,4 +168,8 @@ -(IBAction)cancelTest:(id)sender{
                             inView:self];
 }
 
+- (void)interruptTestUI {
+    [self.runningTestsLabel setText:NSLocalizedString(@"Dashboard.Running.Stopping.Title", nil)];
+    [self.logLabel setText:NSLocalizedString(@"Dashboard.Running.Stopping.Notice", nil)];
```
Comment (bassosimone) [inline ooniprobe/View/RunTest/TestRunningViewController.m:173]: shouldn't you remove the observer here?

**Review by bassosimone — CHANGES_REQUESTED** [review]
Thank you! I think we need a documenting comment and, also, there is a place where I am in doubt of whether we need to delete an observer. 🧇

File: ooniprobe/AppDelegate.m (hunk 1 above)
Comment (lorenzoPrimi) [reply ooniprobe/AppDelegate.m:32]: Yes, to create the singleton. Just the comment will be enough?

File: ooniprobe/View/RunTest/TestRunningViewController.m (hunk 2 above)
Comment (lorenzoPrimi) [reply ooniprobe/View/RunTest/TestRunningViewController.m:173]: In theory I should remove the observer when the View  is dismissed. Good catch

Comment (lorenzoPrimi) [reply ooniprobe/View/RunTest/TestRunningViewController.m:173]: Bot on the other hand, this is automatically done by the OS when it's destroyed

File: ooniprobe/AppDelegate.m (hunk 1 above)
Comment (bassosimone) [reply ooniprobe/AppDelegate.m:33]: Absolutely, thank you!

File: ooniprobe/View/RunTest/TestRunningViewController.m (hunk 2 above)
Comment (bassosimone) [reply ooniprobe/View/RunTest/TestRunningViewController.m:173]: Okay, thanks for clarifying!

**Review by bassosimone — APPROVED** [review]
🐳

**Pull request merged**